examples/aa-simple-dapp/src/hooks/usePluginManager.ts

```diff
@@ -0,0 +1,146 @@
+import { chain } from "@/config/client";
+import {
+  IAccountLoupe,
+  MSCA,
+  MultiOwnerPlugin,
+  SessionKeyPlugin,
+  TokenReceiverPlugin,
+  installPlugin as aaInstallPlugin,
+  uninstallPlugin as aaUninstallPlugin,
+  encodeFunctionReference,
+} from "@alchemy/aa-accounts";
+import { AlchemyProvider } from "@alchemy/aa-alchemy";
+import { useCallback, useMemo, useState } from "react";
+import { Address, encodeAbiParameters, parseAbiParameters } from "viem";
+import { useAsyncEffect } from "./useAsyncEffect";
+
+export enum PluginType {
+  MULTI_OWNER = "Multi Owner Plugin",
+  TOKEN_RECEIVER = "Token Receiver Plugin",
+  SESSION_KEY = "Session Key Plugin",
+}
+
+export const usePluginManager = ({
+  scaAddress,
+  provider,
+}: {
+  scaAddress: Address | undefined;
+  provider: AlchemyProvider;
+}) => {
+  const [installedPlugins, setInstalledPlugins] = useState<
+    ReadonlyArray<Address>
+  >([]);
+
+  const tokenReceiverEnabled = useMemo(
+    () =>
+      installedPlugins.includes(
+        TokenReceiverPlugin.meta.addresses[provider.rpcClient.chain.id]
+      ),
+    [installedPlugins, provider.rpcClient.chain.id]
+  );
+
+  const sessionKeyEnabled = useMemo(
+    () =>
+      installedPlugins.includes(
+        SessionKeyPlugin.meta.addresses[provider.rpcClient.chain.id]
+      ),
+    [installedPlugins, provider.rpcClient.chain.id]
+  );
+
+  const installPlugin = useCallback(
+    async (type: PluginType, params?: any) => {
+      if (scaAddress == null || !provider.isConnected<MSCA>()) {
+        return;
+      }
+
+      switch (type) {
+        case PluginType.SESSION_KEY:
+          return aaInstallPlugin(provider, {
```

Comment:
this should be greatly simplified by the methods generated on `SessionKeyPlugin` and `TokenReceiverPlugin` for installs. might make sense to try and condense this stuff by leveraging the `usePlugin` hook in this hook for each plugin you want to support. will be good dogfooding of those utils